test/operator.jl

```diff
@@ -586,6 +599,42 @@ const sub2 = JuMP.repl[:sub2]
                                          x[2] + 3x[3]])
 
         @test vec_eq(@JuMP.Expression(A*x/2), A*x/2)
+        @test vec_eq(X*v,  [4X11; 6X23; 0])
+        @test vec_eq(v'*X,  [4X11  0   5X23])
+        @test vec_eq(v.'*X, [4X11  0   5X23])
+        @test vec_eq(X'*v,  [4X11;  0;  5X23])
+        @test vec_eq(X.'*v, [4X11; 0;  5X23])
+        @test vec_eq(X*A,  [2X11  X11  0
+                            0     X23  2X23
+                            0     0    0   ])
+        @test vec_eq(A*X,  [2X11  0    X23
+                            X11   0    2X23
+                            0     0    X23])
+        @test vec_eq(A*X', [2X11  0    0
+                            X11   X23  0
+                            0     2X23 0])
+        @test vec_eq(X'*A, [2X11  X11  0
+                            0     0    0
+                            X23   2X23 X23])
+        @test vec_eq(X.'*A, [2X11 X11  0 
+                             0    0    0
+                             X23  2X23 X23])
+        @test vec_eq(A'*X, [2X11  0 X23
+                            X11   0 2X23
+                            0     0 X23])
+        @test vec_eq(X.'*A, X'*A)
+        @test vec_eq(A.'*X, A'*X)
+        @test vec_eq(X*A, X*B)
+        @test vec_eq(Y'*A, Y.'*A)
+        @test vec_eq(A*Y', A*Y.')
+        @test vec_eq(Z'*A, Z.'*A)
+        @test vec_eq(Xd'*Y, Xd.'*Y)
+        @test vec_eq(Y'*Xd, Y.'*Xd)
+        @test vec_eq(Xd'*Xd, Xd.'*Xd)
+        # @test_broken vec_eq(A*X, B*X)
+        # @test_broken vec_eq(A*X', B*X')
+        @test vec_eq(X'*A, X'*B)
+        # @test_broken(X'*X, X.'*X) # sparse quadratic known to be broken
```

Comment:
It is part of a plethora of existing problems with sparse-matrix multiplication, which really needs to be reimplemented (see #907, which still has me confused).  I haven't made an issue for this particular problem.  Give me a few minutes to go back to master and check the error it throws, and I'll create a new issue for it.

Sorry my comments aren't better, I'm a physicist rather than a software engineer by training, so even this is one of the most formal code-writing processes I've ever been through!  I'll try to keep things a little clearer.